routes/web.php

```diff
@@ -807,6 +807,7 @@
         });
         Route::get('gallery', ['as' => 'gallery', 'uses' => 'AchievementController@gallery']);
     });
+    Route::get('achieve/{achievement}', ['as' => 'achieve', 'uses' => 'AchievementController@achieve']);
```

Comment:
There is no auth check in the controller, so you should use the auth middleware here.